catalog/core/catalog-core-solr/src/main/java/ddf/catalog/source/solr/json/MetacardTypeDeserializer.java

```diff
@@ -0,0 +1,67 @@
+/**
+ * Copyright (c) Codice Foundation
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package ddf.catalog.source.solr.json;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+import ddf.catalog.data.AttributeDescriptor;
+import ddf.catalog.data.AttributeType;
+import ddf.catalog.data.MetacardType;
+import ddf.catalog.data.impl.AttributeDescriptorImpl;
+import ddf.catalog.data.impl.BasicTypes;
+import ddf.catalog.data.impl.MetacardTypeImpl;
+import java.io.IOException;
+import java.util.HashSet;
+import java.util.Iterator;
+import java.util.Set;
+
+public class MetacardTypeDeserializer extends StdDeserializer<MetacardType> {
+
+  public MetacardTypeDeserializer() {
+    super(MetacardType.class);
+  }
+
+  @Override
+  public MetacardType deserialize(
+      JsonParser jsonParser, DeserializationContext deserializationContext)
+      throws IOException, JsonProcessingException {
+    JsonNode node = jsonParser.getCodec().readTree(jsonParser);
+
+    String name = node.get("name").textValue();
+
+    Set<AttributeDescriptor> attributeDescriptors = new HashSet<>();
+
+    Iterator<JsonNode> i = node.get("attributeDescriptors").elements();
+    while (i.hasNext()) {
+      JsonNode adNode = i.next();
+      String attributeName = adNode.get("name").textValue();
```

Comment:
❓ There is an assumption that all the nodes are present in the Json. If not then NullPointerExceptions would occur as `null` can be returned from `JsonNode.get()`.